src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -62,26 +62,28 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String idTokenString = null;
     while(headers.hasMoreElements()){
         idTokenString = headers.nextElement();
+         //header passed in should always have ONLY 1 element (just the id token)
+         //therefore, id token should always be the last element.
```

Comment:
We don't know that'll always be true. The website might change in the future and you might want to send more headers to the servlet. Is there any way we can check and be sure that we're getting the id token and not anything else without relying on position?

Reply:
ok (sorry its been a sec since i wrote this code) I just realized that by doing request.getHeaders("idtoken") its always just going to return the value of the header labelled idtoken. U have to still iterate through it, but I'll update the comment to reflect this